libcxx/test/std/utilities/utility/pairs/pair.astuple/get_ref.pass.cpp

```diff
@@ -0,0 +1,68 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11
+
+// <utility>
+
+// template <class T1, class T2> struct pair
+
+// template<size_t I, class T1, class T2>
+//     typename tuple_element<I, std::pair<T1, T2> >::type&&
+//     get(pair<T1, T2>&&);
+
+#include <cassert>
+#include <utility>
+
+#include "test_macros.h"
+
+TEST_CONSTEXPR_CXX14 bool test() {
+  int i = 1;
+  int j = 2;
+
+  {
+    std::pair<int&, int&&> p(i, std::move(j));
```

Comment:
Maybe test with entirely different types too?

Reply:
The point here is to make sure we're forwarding references correctly, so I'm not sure it makes much of a difference testing with different types.

Reply:
This still needs to be done.